libs/service-portal/access-control/delegations/src/hooks/useDomains.ts

```diff
@@ -80,25 +80,50 @@ export const useDomains = (includeDefaultOption = true) => {
     sessionStore.setItem('domain', name)
 
     const query = new URLSearchParams(location.search)
+    const currentDomain = query.get('domain')
 
-    if (query.get('domain')) {
+    if (currentDomain && currentDomain !== displayNameQueryParam) {
```

Comment:
~This condition confuses me. Looks like it's comparing the domain query parameter to the domain query parameter, loaded in different ways. Why?

And is `displayNameQueryParam` an old variable name?

Reply:
This is to make sure that we are not stuck in an endless useEffect loop.
Once the history push or replace runs then the hook automatically rerenders causing endless loop